### PR TITLE
dummy ad hoc changes to illustrate colorscale flipping

### PR DESCRIPTION
**Pull request by remicousin**
This is not a PR meant to be merged. Rather to more easily describe changes to support discussion for this [issue](https://github.com/iridl/python-maprooms/issues/146). The change Aaron pointed to is one thing but there is also a change eliminating DataArrayEntry which makes changed necessary in both maproom.py and pingrid.py to try to repeat this past behavior.

The minimalist changes I've done intend to reproduce what was happening before using the correlationcolorscale case study. So when running the app, change to Precipitation Anomaly to fall into that case. Those changes revert the colorscale for both the colorscale and its application to the data on the map. However, even though one can see that that correlation colorscale is now in the right order (from blue to red), there seems to be 2 wrong individual colors: the central neutral color which is
/moccasin       255 228 181     RGBdef
in Ingrid (some sort of beige color). And the following color starting the "reds" should be yellow and it looks like some pale green. [Reminder](http://iridl.ldeo.columbia.edu/ds%3A/SOURCES/.NOAA/.NCEP-NCAR/.CDAS-1/.MONTHLY/.Intrinsic/.PressureLevel/.phi/P/(250)/VALUE/dup/T/(Jan%201991)/(Dec%202020)/RANGE/yearly-climatology/sub/std_anomaly_colors2//long_name/(Geopotential%20Height%20Anomaly)/def//name/(phi1)/def/SOURCES/.NOAA/.NCEP-NCAR/.CDAS-1/.MONTHLY/.Intrinsic/.PressureLevel/.phi/P/(250)/VALUE/T/(Jan%201991)/(Dec%202020)/RANGE/yearly-climatology/DATA/60/STEP//name/(phiclim)/def/%3Ads/a-/.phi1/correlationcolorscale/DATA/AUTO/AUTO/RANGE/-a-/.phiclim/-a/X/Y/fig%3A/colors/grey/unlabelledcontours/black/thin/solid/coasts/countries/%3Afig/(phi1)/-200/200/plotrange//T/last/plotvalue/X/-20/340/plotrange/Y/-65/75/plotrange//plotaxislength/650/psdef//plotborder/72/psdef//XOVY/null/psdef//color_smoothing/null/psdef//antialias/true/psdef/#expert) of what Ingrid's colorscale looks like.

So I'd think something is off in the color reverting as I applied it. Either what I did did reproduce what was happening before and there has always been a problem with that very case (that was either ignored or undiscovered).

Given then many things have changed since then in pingrid.py. It may not be worth to try and understand the changes that matter to this problem since then, and rather either reinstate a new reverting option or abandon trying to convert ingrid colorscales to python.

### CODE REVIEW

**Comment by aaron-kaplan** [comment]
> This other dummy change prints the right colorscale from Ingrid

Cool, sounds like we've gotten to the bottom of it? Did you verify all the colormaps we use, or just one?

**Comment by remicousin** [comment]
I can test the other ingrid colorscales in forecast.

Then is pingrid.RAINBOWCOLORMAP an ingrid one or other? I am not sure what is used in fbf. Then parse_color, that I changed, is used by cascading functions, so I am not sure where the proper changes should be. I guess sometimes we are feeding a RGB series and sometimes a BGR and Igor had a check on that somewhere to apply the switch or not, with his reverseclormap boolean. But that's in a function that is gone.

I don't know if it's any clearer to you. Just checking before trying and going to the bottom of things.

**Comment by aaron-kaplan** [comment]
I don't remember, but I think RAINBOW_COLORMAP probably came from Ingrid. That's circumstantially supported by the fact that there's one called rainbowColorMap in ingridinit.tex.

The colormaps used by FBF are in the [config file](https://github.com/iridl/python-maprooms/blob/master/fbfmaproom/fbfmaproom-sample.yaml) -- search for &raincm, &pnepcm, and &vulncm.

Looking at the tile generation code, I think rather than introducing the RGB/RGBA classes as you've done here, you should stick with BGR/BGRA, and instead flipt the order of the bytes in `parse_color`. i.e.

```
    return BGRA(v >> 16 & 0xFF, v >> 8 & 0xFF, v >> 0 & 0xFF, 255)

```

**Comment by remicousin** [comment]
hmmm... I also need to know what CONFIG to give to run the tests.

**Comment by remicousin** [comment]
hmmm... I am not quite there yet. It's running it seems but this:
http://devi:8063/fbfmaproom/niger/
returns a JSON (I added the country as without, it returns KeyError: 'path'

I am also not on the post pyaconf era, should I rebase to get going?

**Comment by aaron-kaplan** [comment]
That's weird. I don't know what would cause that. Please do rebase to get us to a good starting place for debugging, but a priori I can't see how that would fix this.

**Comment by remicousin** [comment]
Rebasing and remaking environment from lock allowed me to build and serve. Now it's apparent to me that the colorscales in fbfmaprooms (at least some), need not be reverted (RGB/BGR) like for Ingrid ones. How do we want to go about this?

**Comment by remicousin** [comment]
@aaron-kaplan let me know if now is a good time to resolve this. As a reminder, the issue is that some colorscales are defined as RGB and others as BRG but our code ignores that and assume always one of the 2. So we need either to pick a case and change the colorscales defined according to the other case, either we need code to be told what to do and when.

**Comment by aaron-kaplan** [comment]
> Now it's apparent to me that the colorscales in fbfmaprooms (at least some), need not be reverted (RGB/BGR) like for Ingrid ones.

But the colorscales in fbfmaproom came from ingrid. At least I thought they did. Have you compared pingrid.RAINBOW_COLORMAP with ingrid's rainbowColorMap?

In any case, if you have a way of interpreting colormap representations that works for all ingrid colormaps (not just for one example), then we will simply define that to be the right interpretation, and rewrite the colormaps in pingrid such that they get interpreted as intended by that definition. Just be sure you've got the ingrid end right first.

**Comment by remicousin** [comment]
With the changes in this PR:

- master's pingrid.RAINBOW_COLORMAP is different from Ingrid's. Using Ingrid's does the right thing.
- the 3 colorscales for flex fcst (PO/N/E and correlation) come from Ingrid and are rendered correctly.
- fbfmaproom's prob colorscale goes now from blue to red (instead of red to blue)
- fbfmaproom's vuln colorscale is now shared of blue (instead of shades of red)
- fbfmaproom's prcp colorscale: I don't know how to reveal it.

So in short, all the colorscales I added as direct copies of Ingrid's colormaps are rendered as expected. All pre-existing colorscales are flipped (RGB/GRB)

**Comment by remicousin** [comment]
Shall I rewrite the FbFmaproom colorscales through the same order (BGR?) than the other ones? so that i) we can readily use something from Ingrid and ii) we don't have to bother with cases where we flip or don't?

**Comment by aaron-kaplan** [comment]
Yes, please.

**Comment by remicousin** [comment]
ok. Can you tell me how to see:
prcp colorscale
?

**Comment by aaron-kaplan** [comment]
I don't know that it's being used in fbf. grep for it, and if you don't
find it, just delete it.

**Comment by remicousin** [comment]
Was resolved through PR #244
